website/docs/guides/errors.md

```diff
@@ -0,0 +1,190 @@
+  try {
+    await run(function*() {
+      throw new Error('boom');
+    });
+  } catch(err) {
+    console.log("got error", err.message) // => "got error boom"
+  }
+}
+```
+
+When treating a task as a promise, if the task becomes halted, the promise is
+rejected with a special halt error:
+
+``` typescript
+import { run } from 'effection';
+
+async function runExample() {
+  try {
+    let task = run();
+    task.halt();
+    await task;
+  } catch(err) {
+    console.log("got error", err.message) // => "got error halted"
+  }
+}
+```
+
+## Error propagation
+
+One of the key principles of structured concurrency is that when a child fails,
+the parent should fail as well. In Effection, when we spawn a task, that task
+becomes linked to its parent. When the child task becomes `errored`, it will
+also cause its parent to become `errored`.
+
+This is similar to the intuition you've built up about how synchronous code
+works: if an error is thrown in a function, that error propagates up the stack
+and causes the entire stack to fail, until someone catches the error.
+
+One of the innovations of async/await code over plain promises and callbacks,
+is that you can use regular error handling with `try/catch`, instead of using special
+error handling constructs. This makes asynchronous code look and feel more like
+regular synchronous codes. The same is true in Effection, we can use `try/catch`
+to deal with errors.
+
+``` typescript
+import { main, sleep } from 'effection';
+
+function* tickingBomb() {
+  yield sleep(1000);
+  throw new Error('boom');
+}
+
+main(function*() {
+  try {
+    yield tickingBomb()
+  } catch(err) {
+    console.log("it blew up:", err.message);
+  }
+});
+```
+
+However, note that something interesting happens when we instead `spawn` the
+`tickingBomb` operation:
+
+``` typescript
+import { main } from 'effection';
+import { tickingBomb } from './ticking-bomb';
+
+main(function*() {
+  yield spawn(tickingBomb());
+  try {
+    yield; // sleep forever
+  } catch(err) {
+    console.log("it blew up:", err.message);
+  }
+});
+```
+
+You might be surprised that we do *not* enter the catch handler here! Instead,
+our entire main task just fails. This is by design! We are only allowed to
+catch errors thrown by whatever we yield to directly, not by any spawned
+children or resources running in the background. This makes error handling more
+predictable, since our catch block will not receive errors from any background
+task, we're better able to specify which errors we actually want to deal with.
+
+## Error boundary
+
+If we do want to catch an error from a spawned task (or from a [Resource][]) then
+we need to introduce an intermediate task which allows us to bring the error into
+the foreground. We call this pattern an "error boundary":
+
+``` typescript
+import { main } from 'effection';
+import { tickingBomb } from './ticking-bomb';
+
+main(function*() {
+  try {
+    yield function*() { // error boundary
```

Comment:
Take this one with a grain of salt because I'm not sure how I feel about it myself. The example as-is is very clear, but it is also a good habit to get into to name your error boundary so that it shows up in the inspector.

```suggestion
    yield function* errorBoundary() {
```

Reply:
I see your point, but I almost feel that this would be confusing, both in the example, and showing up in the inspector. Maybe a more neutral name like `inner` or something like that would make sense, but if I saw this in the inspector I would be confused why everything is under an error boundary.